pkg/bundle/bundle.go

```diff
@@ -69,99 +77,82 @@ type bundle struct {
 
 	// Options holds options for the Bundle controller.
 	Options
+
+	// patchResourceOverwrite allows use to override the patchResource function
+	// it is used for testing purposes
+	patchResourceOverwrite func(ctx context.Context, obj interface{}) error
 }
 
 // Reconcile is the top level function for reconciling over synced Bundles.
 // Reconcile will be called whenever a Bundle event happens, or whenever any
 // related resource event to that bundle occurs.
 func (b *bundle) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	result, statusPatch, resultErr := b.reconcileStatusPatch(ctx, req)
+	if statusPatch != nil {
+		con, patch, err := ssa_client.GenerateBundleStatusPatch(req.Name, req.Namespace, statusPatch)
+		if err != nil {
+			err = fmt.Errorf("failed to generate connection patch: %w", err)
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{resultErr, err})
+		}
+
+		if err := b.client.Status().Patch(ctx, con, patch, &client.SubResourcePatchOptions{
+			PatchOptions: client.PatchOptions{
+				FieldManager: "trust-manager",
+				Force:        ptr.To(true),
+			},
+		}); err != nil {
+			err = fmt.Errorf("failed to apply connection patch: %w", err)
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{resultErr, err})
+		}
+	}
+
+	return result, resultErr
+}
+
+func (b *bundle) reconcileStatusPatch(ctx context.Context, req ctrl.Request) (result ctrl.Result, bundlePatch *trustapi.BundleStatus, returnedErr error) {
```

Comment:
```suggestion
func (b *bundle) reconcileBundle(ctx context.Context, req ctrl.Request) (result ctrl.Result, statusPatch *trustapi.BundleStatus, returnedErr error) {
```

WDYT? It does a bit more than reconciling status patch?